i6_models/parts/mha.py

```diff
@@ -0,0 +1,80 @@
+__all__ = ["MultiheadAttentionV1", "MultiheadAttentionV1Config"]
+
+import math
+
+import torch
+from torch import nn
+from i6_models.config import ModelConfiguration
+from i6_models.util import compat
+from dataclasses import dataclass
+
+
+@dataclass
+class MultiheadAttentionV1Config(ModelConfiguration):
+    input_dim: int
+    num_att_heads: int
+    att_weights_dropout: float
+    dropout: float
+
+    def __post_init__(self) -> None:
+        super().__post_init__()
+        assert self.input_dim % self.num_att_heads == 0, "input_dim must be divisible by num_att_heads"
+
+
+class MultiheadAttentionV1(nn.Module):
+    def __init__(self, cfg: MultiheadAttentionV1Config):
+        super().__init__()
+        self.cfg = cfg
+        self.num_att_heads = cfg.num_att_heads
+        self.input_dim = cfg.input_dim
+        self.dim_heads = self.input_dim // self.num_att_heads
+        self.layernorm = torch.nn.LayerNorm(cfg.input_dim)
+
+        self.out_proj = torch.nn.Linear(in_features=cfg.input_dim, out_features=cfg.input_dim, bias=True)
+        self.in_proj = torch.nn.Linear(in_features=cfg.input_dim, out_features=3 * cfg.input_dim, bias=True)
+
+        self.norm = math.sqrt(self.input_dim)
+        self.softmax = nn.Softmax(-1)
+        self.dropout = nn.Dropout(cfg.att_weights_dropout)
+
+    def forward(
+            self,  
+            query: torch.Tensor,
+            key: torch.Tensor,
+            value: torch.Tensor,
+            key_padding_mask: torch.Tensor,
+            need_weights: bool):
```

Comment:
It would be nice to have the inputs and expected shapes commented in a docstring

Reply:
Added.